File: moderate/array-absurdity/array.py
import sys
test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    line = test.strip().split(';')
    lst = [int(x) for x in line[1].split(',')]
    for i, ele in enumerate(lst):
        if ele in lst[i+1:]:
            print ele
            break

test_cases.close()

File: moderate/array-absurdity/array.pl
open(INFILE, "<$ARGV[0]") or die("Cannot open file $ARGV[0] for reading: $!");
while(my $line = <INFILE>) {
    next if($line =~ m/^s$/); # skip blank lines
    chomp($line);
    @vals = split(';', $line);
    @ints = split(',', $vals[1]);
    %st = ();
    $st{$_}++ for @ints;
    @vals = sort {$st{$b}<=>$st{$a}} keys %st;
    print "$vals[0]\n";
}
close(INFILE);
